src/sentry/eventstream/base.py

```diff
@@ -22,6 +32,16 @@ class ForwarderNotRequired(NotImplementedError):
     """
 
 
+class GroupState(TypedDict):
+    is_new: bool
+    is_regression: bool
+    is_new_group_environment: bool
+    primary_hash: Optional[str]
+
+
+GroupStates = Mapping[int, GroupState]
```

Comment:
What's the int in the key going to be? Is that meant to be the group ID?

Reply:
The int key is the group_id.

I was thinking about this and maybe it makes more sense to have something like this:

```
class GroupState(TypedDict):
  id: int
  is_new: bool
  is_regression: bool
  is_new_group_environment: bool
  primary_hash: Optional[str]

GroupStates = Sequence[GroupState]
```

Reply:
This seems more straightforward to me than keeping the sequence of group IDs in one place and the mapping of ID to the other data in another.

Reply:
Not sure if this will be useful in the short term, but another positive is the list preserves the ordering.

Will make the change.